calculator_test.go

```diff
@@ -2,23 +2,310 @@ package calculator_test
+			a:    .5,
+			b:    -40,
+			want: -20,
+			f:    calculator.Multiply,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.f(tc.a, tc.b)
+			if tc.want != got {
+				t.Errorf("want %f, got %f", tc.want, got)
+			}
+		})
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name        string
+		a, b, want  float64
+		errExpected bool
+	}{
+		{
+			name:        "Divide 2 and 2",
+			a:           2,
+			b:           2,
+			want:        1,
+			errExpected: false,
+		},
+		{
+			name:        "Divide a positive and a negative",
+			a:           1,
+			b:           -1,
+			want:        -1,
+			errExpected: false,
+		},
+		{
+			name:        "Divide a smaller number by a larger number",
+			a:           10,
+			b:           100,
+			want:        .1,
+			errExpected: false,
+		},
+		{
+			name:        "Divide a negative integer by a positive decimal",
+			a:           -40,
+			b:           .5,
+			want:        -80,
+			errExpected: false,
+		},
+		{
+			name:        "Divide a positive integer by zero",
+			a:           6,
+			b:           0,
+			want:        0,
+			errExpected: true,
+		},
+		{
+			name:        "Divide zero by an integer",
+			a:           0,
+			b:           10,
+			want:        0,
+			errExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		got, err := calculator.Divide(tc.a, tc.b)
+		errReceived := err != nil
+		if tc.errExpected != errReceived {
+			t.Fatalf("Unexpected error status: %v", err)
+		}
+
+		if !tc.errExpected && tc.want != got {
+			t.Errorf("want %f, got %f", tc.want, got)
+		}
+	}
+}
+
+func TestAddRandom(t *testing.T) {
+	t.Parallel()
+	iterations := 100
+	for i := 0; i < iterations; i++ {
+		a := rand.Float64() * 10
+		b := rand.Float64() * 10
+
+		sum := a + b
+
+		got := calculator.Add(a, b)
+		if sum != got {
+			t.Errorf("Expected sum of %f and %f to match", a, b)
+			t.Errorf("Got %f and %f (the latter from Add)", sum, got)
+		}
+	}
+}
+
+func TestSqrRoot(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name        string
+		a, want     float64
+		errExpected bool
+	}{
+		{
+			name:        "Square root of 4",
+			a:           4,
+			want:        2,
+			errExpected: false,
+		},
+		{
+			a:           16,
+			want:        4,
+			name:        "Square root of 16",
+			errExpected: false,
+		},
+		{
+			name:        "Square root of 100",
+			a:           100,
+			want:        10,
+			errExpected: false,
+		},
+		{
+			name:        "Square root of -20",
+			a:           -20,
+			errExpected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := calculator.SqrRoot(tc.a)
+			errReceived := err != nil
+			if tc.errExpected != errReceived {
+				t.Fatalf("Unexpected error status: %v", err)
+			}
+
+			if !tc.errExpected && tc.want != got {
+				t.Errorf("want %f, got %f", tc.want, got)
+			}
+		})
+	}
+}
+
+func TestVariadicFunctions(t *testing.T) {
```

Comment:
Try integrating this with your multi-function test (so that some cases have extra parameters, and some don't).